### DISCUSSION TITLE
A feature to avoid IDE extensions and build them in cli

### DISCUSSION THREAD

**Discussion by devopstoday11** [charmbracelet/bubbletea#947] [Q&A]
Is there already an effort going on or already done, Or can we plan for contribution for the below?

This is a feature request! 

I'll try to explain with an example.

If you take Tabnine or Codeium or CodiumAI as references, those are installed/added as IDE extensions in IDEs like Zed, VScode, JetBrains, etc. At a high level they basically give 4 categories in their interface, among many others. 
- a chat 
- a place to add an API token to provide authorization to connect and use their SaaS backend services 
- executing commands in command line 
- selecting some lines or text of  code in a file and then right click on the selection to do perform actions, like generate unit test for the selection function or something else.

Is it possible to build such features in a cli tool that's built using bubbletea TUI framework? If yes, how?

**Comment by bashbunni** [comment]
Hey @devopstoday11 thanks for the idea! I'd love to hear more to fully understand what you mean. I'm having difficulty understanding a few things in your post, if you don't mind clarifying that would be very much appreciated!

Do you have any examples of what this interface might look like? images of what you want to emulate with Bubble Tea? Also, what kind of changes are you thinking to the Library itself or is this more of an idea of something to build with Bubble Tea?

**Reply by devopstoday11** [reply]
@bashbunni 
I will try to explain with some screenshots I took from my VScode IDE: 
![image](https://github.com/charmbracelet/bubbletea/assets/59978378/b4f896f2-ae70-44b6-9312-c4a8770aeb25)
- Using the first image as reference, the way how chat is available to talk to LLM. Is it possible to build such chat interface within the CLI tool itself by using Bubbletea? 

# 
- If I have a necessity to save LLM API token securely in the CLI tool locally, how to build such an option as an interface?
# 

![image](https://github.com/charmbracelet/bubbletea/assets/59978378/c3dc9a0a-a2f1-400e-946d-d77d01235f56)
- In this second image, using the mouse, I selected some lines.  Is it possible to add mouse interactions to the CLI tool using Bubbletea?
# 
![image](https://github.com/charmbracelet/bubbletea/assets/59978378/afa58cc2-40c6-40d2-8ac5-d3c1ad3a1c66)
- After selection some lines, I right clicked on my mouse and then clicked on the option I want.  Is it possible to build such interface within CLI tool using Bubbletea to give such " lines selection and triggering some action(s) for the selected lines "? 
# 
- Maybe this should be the first question to ask :smiley:  :- 
Is it possible to build an interface in a CLI tool using Bubbletea to open a project folder, or some files, or one file, select lines, and trigger some actions?

**Reply by devopstoday11** [reply]
@bashbunni Did my explanation help get the idea?

**Comment by bashbunni** [comment]
I'm actually going to move this to a discussion as that's an easier format for having a proper discussion about this. I can convert it back to an issue once there's enough context to take action :)
